syntax="proto3";

package account.error;

enum AccountErrorCode {
    ERROR_UNSPECIFIED = 0;
    ERROR_INVALID_PARAMETER = 1;
    ERROR_RESOURCE_NOT_FOUND = 2;
    ERROR_INVALID_ENTITY_TYPE = 3;
    ERROR_ENTITY_USER_NOT_FOUND = 4;
    ERROR_INVALID_PERMISSION_ROLE = 5;
    ERROR_OPERATION_FAILED = 6;

    // additional errors start at value 10
    ERROR_EMAIL_ALREADY_REGISTERED = 10;
}